### PR TITLE
feat(snomed): introduce NamespaceIdProvider to extract namespace values.

### PR DESCRIPTION
**Pull request by cmark**
...from a collection of namespaceConceptIds

Introduce namespaceConceptId in SNOMED CT component create endpoints to
support creation of SNOMED CT content by specifying the
namespaceConceptId as opposed to the actual namespace value. The system
will be able to extract this information from the namespace concept and
use it to generate the new IDs with the correct namespace.

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/NamespaceConceptIdStrategy.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	private static final long serialVersionUID = 1L;
+	
+	private final String namespaceConceptId;
+
+	public NamespaceConceptIdStrategy(String namespaceConceptId) {
+		this.namespaceConceptId = checkNotNull(namespaceConceptId);
+	}
+	
+	@Override
+	public String getNamespace() {
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/NamespaceConceptIdStrategy.java:36]: There is one additional call to this method that might not like the change in meaning:

https://github.com/b2ihealthcare/snow-owl/blob/98bf7610eb0db60d63aa887ac9a23b3a148d7f54/snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptCreateRequestBuilder.java#L49

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/NamespaceIdProvider.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		/* 
+		 * The International core namespace concept will not have an FSN matching the pattern,
+		 * so remove it from the set, and convert it to the empty namespace directly. 
+		 */
+		if (mutableNamespaceConceptIds.remove(Concepts.CORE_NAMESPACE)) {
+			namespacesByNamespaceConceptId.put(Concepts.CORE_NAMESPACE, "");
+		}
+		
+		// Find the FSN of namespace SCTIDs
+		SnomedRequests.prepareSearchDescription()
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/NamespaceIdProvider.java:68]: An `id`, `conceptId`, `term` field selection might speed up things further, although we only expect to retrieve a small number of descriptions.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/NamespaceConceptIdStrategy.java (hunk 1 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/NamespaceConceptIdStrategy.java:36]: Fixed.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/NamespaceIdProvider.java (hunk 2 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/NamespaceIdProvider.java:68]: Done.

**Review by apeteri — APPROVED** [review]
👍

**Review by nagyo — APPROVED** [review]
LGTM!

**Pull request merged**